### PR TITLE
Compile READMEs to proper markdown

### PR DESCRIPTION
**Pull request by igloosi**
This PR
adds a gulp plugin to compile to markdown
amends generate:readme task to add a pipe step

### CODE REVIEW

File: tasks/gulp/generate-readme.js [hunk 1]
```diff
@@ -34,6 +35,7 @@ gulp.task('generate:readme', () => {
     path: [paths.src + 'views', paths.components],
     manageEnv: manageEnvironment
   }))
+  .pipe(toMarkdown({ gfm: true }))
```
Comment (gemmaleigh) [inline tasks/gulp/generate-readme.js]: Can we add a comment here about where this came from and what it does.

**Comment by gemmaleigh** [comment]
This is awesome and addresses @36degrees' concern about being able to read the markdown files in your editor. 

We'll still need to address #226, as we've got generated files in `/src/`, but while we're likely to be changing the content for many of these files - keeping the task to generate the documentation from a components source files will make it easier to keep component documentation up-to-date.

**Comment by igloosi** [comment]
opened issue #226 for that

**Comment by gemmaleigh** [comment]
:1st_place_medal:

**Comment by 36degrees** [comment]
👏

**Pull request merged**